R/join_keys-c.R

```diff
@@ -10,7 +10,7 @@
 #'   jk,
 #'   join_keys(
 #'     join_key("ds4", keys = c("pk4", "pk4_2")),
-#'     join_key("ds4", "ds3", c(pk4_2 = "pk3"))
+#'     join_key("ds3", "ds4", c(pk3 = "pk4_2"))
 #'   )
 #' )
 c.join_keys <- function(...) {
```

Comment:
Is there a reason keys are concatenated `first < add (empty < add second < add third < ...)`? I naively think all of `...` can be put into the `Reduce` but I suppose I'm missing something.

Reply:
@insightsengineering/nest-core-dev any thoughts on whether we should change this behavior?

- as is (restrictive)
- unrestrictive for `c(...)` and `join_keys(...)`
- unrestrictive for `c(..)` call only

Reply:
I think both `c(...)` and `join_keys(...)` should be unrestricted. or atleast for `c(...)` as its confusing to see error for more then two datasets as [described here](https://github.com/insightsengineering/teal.data/pull/215#discussion_r1432806727).

Reply:
I'll make the proposed change @kartikeyakirar , that's a good argument :+1:

Reply:
Please check b3177082